lib/response_comparator.rb

```diff
@@ -21,15 +21,35 @@ def initialize(primary_response, secondary_response, full_comparison_pct = 0)
   def compare
     start = Time.now
     comparison = quick_comparison
-    comparison.merge!(differences) if @secondary_response && full_comparison?(comparison, full_comparison_pct)
+    comparison.merge!(differences) if secondary_response_completed_ok? && full_comparison?(comparison, full_comparison_pct)
     comparison[:comparison_time_seconds] = Time.now - start
     comparison
   end
 
   def quick_comparison
+    secondary = if secondary_response_completed_ok?
+                  response_stats(@secondary_response)
+                else
+                  error_fields(@secondary_response)
+                end
+
     {
       primary_response: response_stats(@primary_response),
-      secondary_response: @secondary_response ? response_stats(@secondary_response) : nil,
+      secondary_response: secondary,
+    }
+  end
+
+  def secondary_response_completed_ok?
+    !@secondary_response.class.ancestors.include?(StandardError)
```

Comment:
If the intention here is to check if @secondary_response is a subclass of StandardError, I think this will do the same thing: `!@secondary_response.is_a?(StandardError)`

Reply:
That was the first version I tried, and I couldn't get it to work consistently across the running appliation and the test cases, so I went with the `.ancestors.include?` version (which did)